types/main.d.ts

```diff
@@ -0,0 +1,624 @@
+declare module "pdfjs" {
+    export class Image {
+        /**
+         * Creates a new image that can be added to one or multiple documents.
+         * @param src a buffer of the image source (supported formats: jpeg, pdf)
+         */
+        constructor(src: Buffer | ArrayBuffer);
+    }
+
+    export class Font {
+        /**
+         * Creates a new AFM font pr OTF font object that can be used with PDF documents.
+         * TFont objects can be used multiple times.
+         * @param arg the font data, as either Buffer or ArrayBuffer of an OTF font or a JSON object descriping an AFM font
+         */
+        constructor(arg: Buffer | ArrayBuffer | FontDescription);
+    }
+
+    export type FontDescription = {
+        fontName: string;
+        fullName: string;
+        familyName: string;
+        italicAngle: number;
+        characterSet: string;
+        fontBBox: number[];
+        underlinePosition: number;
+        underlineThickness: number;
+        kerning: any;
+        widths: (number | null)[];
+    };
+
+    export const mm: number;
+    export const cm: number;
+
+    export type DocumentProperties = {
+        /**
+         * The document's title
+         */
+        title?: string;
+        /**
+         * The name of the person who created the document
+         */
+        author?: string;
+        /**
+         * The subject of the document
+         */
+        subject?: string;
+        /**
+         * Keywords associated with the document
+         */
+        keywords?: string;
+        /**
+         * If the document was converted to PDF from another format, the name of the conforming product that created the original document from which it was converted
+         */
+        creator?: string;
+        /**
+         * If the document was converted to PDF from another format, the name of the conforming product that converted it to PDF
+         * default: pdfjs v1.3 (github.com/rkusa/pdfjs)
+         */
+        producer?: string;
+        /**
+         * The date and time the document was created
+         * default: new Date()
+         */
+        creationDate?: Date;
+        /**
+         * The date and time the document was most recently modified
+         */
+        modDate?: Date;
+    };
+
+    export class Document extends Fragment {
+        /**
+         * Creates a new PDF document
+         * @param opts document options
+         */
+        constructor(opts: DocumentOptions);
```

Comment:
Another TODO for myself (no action required here). I think `opts` shouldn't be required, `opts.font` should rather default to e.g. `Helvetica`